drivers/usb/device/usb_dc_stm32.c

```diff
@@ -992,6 +992,17 @@ int usb_dc_detach(void)
 	 */
 #endif /* CONFIG_SOC_SERIES_STM32WBX */
 
+	LOG_DBG("HAL_PCD_DeInit");
+	status = HAL_PCD_DeInit(&usb_dc_stm32_state.pcd);
+	if (status != HAL_OK) {
+		LOG_ERR("PCD_DeInit failed, %d", (int)status);
+		return -EIO;
+	}
+
+	if (irq_is_enabled(USB_IRQ)) {
+		irq_disable(USB_IRQ);
+	}
+
```

Comment:
Looks good, but don't we need to undo everything done in attach(), like disabling clocks enabled by usb_dc_stm32_clock_enable().